cypress/integration/home_page_test.js

```diff
@@ -0,0 +1,99 @@
+      cy.fixture('triggers').as('triggers');
+      cy.server();
+      cy.route('GET', '/appengine/v1/*/stats/devices', '@devicesStats');
+      cy.route('GET', '/realmmanagement/v1/*/triggers', '@triggers');
+      cy.route('GET', '/realmmanagement/v1/*/interfaces', '@interfaces');
+      cy.route('GET', '/realmmanagement/v1/*/interfaces/*', '@interfaceMajors');
+      cy.route('GET', '/realmmanagement/v1/*/interfaces/*/*', '@interface');
+      cy.route('GET', '/appengine/health', '');
+      cy.route('GET', '/realmmanagement/health', '');
+      cy.route('GET', '/pairing/health', '');
+      cy.route('GET', '/flow/health', '');
+      cy.login();
+    });
+
+    it('successfully loads the Home page after logging in', () => {
+      cy.location('pathname').should('eq', '/');
+      cy.get('h2').contains('Astarte Dashboard');
+    });
+
+    it('has an API Status card with a table to display services health', () => {
+      cy.get('.card')
+        .contains('API Status')
+        .parents('.card')
+        .within(() => {
+          const services = ['Realm Management', 'AppEngine', 'Pairing', 'Flow'];
+          cy.get('table tbody').find('tr').should('have.length', services.length);
+          services.forEach((service, index) => {
+            cy.get(`table tbody tr:nth-child(${index + 1})`).within(() => {
+              cy.contains(service);
+              cy.contains('This service is operating normally');
+            });
+          });
+        });
+    });
+
+    it('has a Devices card that shows stats on devices', function () {
+      cy.get('.card')
+        .contains('Devices')
+        .parents('.card')
+        .within(() => {
+          cy.contains('Registered devices').next().contains(this.devicesStats.data.total_devices);
+          cy.contains('Connected devices')
+            .next()
+            .contains(this.devicesStats.data.connected_devices);
+        });
+    });
+
+    it('has a Interfaces card that shows available interfaces', function () {
+      cy.get('.card')
+        .contains('Interfaces')
+        .parents('.card')
+        .within(() => {
+          this.interfaces.data.forEach((interfaceName) => {
+            cy.contains(interfaceName);
+          });
+          cy.get('button').contains('Install a new interface');
+          cy.contains(this.interface.data.interface_name).click();
+        });
+      cy.location('pathname').should(
+        'eq',
+        `/interfaces/${this.interface.data.interface_name}/${this.interface.data.version_major}`,
+      );
+      cy.visit('/');
+      cy.get('.card button').contains('Install a new interface').click();
+      cy.location('pathname').should('eq', '/interfaces/new');
+    });
+
+    it('has a Triggers card that shows available triggers', function () {
+      cy.get('.card')
+        .contains('Triggers')
+        .parents('.card')
+        .within(() => {
+          this.triggers.data.forEach((trigger) => {
```

Comment:
See comment above, this displays at most 5 triggers

Reply:
Nice catch again!